src/Controls/src/Core/VisualStateManager.cs

```diff
@@ -674,10 +674,7 @@ internal static bool HasVisualState(this VisualElement element, string name)
 				for (var j = 0; j < group.States.Count; j++)
 				{
 					VisualState state = group.States[j];
-					if (state.Name == name)
-					{
-						return true;
-					}
+		return state.Name == name;
```

Comment:
This will return `false` on the first non-matching state name.

Reply:
Derp :-/ fixed

Reply:
sorry for the confusion!